src/module.c

```diff
@@ -781,16 +789,23 @@ static inline int add(RedisModuleCtx *ctx,
 int TSDB_madd(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     RedisModule_AutoMemory(ctx);
 
-    if (argc < 4 || (argc - 1) % 3 != 0) {
+    if (argc < 4 || (argc) % 3 == 0) {
```

Comment:
min argc: MADD APPEND_ONLY KEY TS VAL -> 5 arguments
the other check is a bit more complicated because you can have either `1 + 3x` or `2 + 3x`, you need to see if we have `appendOnly` enabled to know which one to use.
But lets first determine if we really need it in MADD.

Reply:
min args is still `4` since `append_only` is optional.
the second check is always used in an unprecise way to eliminate options that are wrong. In TS.CREATE, for example, we just check for the lower bound and not parse each option is used correctly. The precise parsing is done later.